cli/src/main/java/org/jboss/as/cli/CommandLineMain.java

```diff
@@ -151,4 +158,31 @@ private static Properties createLogManagerConfig(final String level) {
         properties.setProperty("formatter.PATTERN.properties", "pattern");
         return properties;
     }
+
+    /**
+     * Creates a file marker under $JBOSS_HOME/bin directory. This is used by the shutdown operation to understand from where we have
+     * launched CLI instance when we are performing an update of a remote installation.
+     */
+    private static void createClientMarker() {
+        try {
+            final String jbossHome = System.getenv("JBOSS_HOME");
```

Comment:
@yersan when CLI is started as a fat jar (bin/client/jboss-cli-client.jar), we don't have this env set. I am wandering if we should attempt to find the jar location (class.getProtectionDomain().getCodeSource().getLocation()) and determinate if that is inside an installation?

Reply:
@jfdenise We should be fine there since Installation Manager high level commands are not available for the CLI if the CLI is not running on a modular environment.